File: src/ros2_bridge/mappers/std_msgs/multi_array_dimension.rs
//! Typed mapper for `std_msgs/msg/MultiArrayDimension`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn multi_array_dimension_to_bus(msg: ros_env::std_msgs::msg::MultiArrayDimension) -> crate::std_msgs::msg::v1::MultiArrayDimension {
    crate::std_msgs::msg::v1::MultiArrayDimension {
        label: crate::ros2_bridge::mappers::convert::from_ros_string(msg.label),
        size: msg.size,
        stride: msg.stride,
    }
}

pub(crate) fn multi_array_dimension_to_ros(bus: crate::std_msgs::msg::v1::MultiArrayDimension) -> ros_env::std_msgs::msg::MultiArrayDimension {
    ros_env::std_msgs::msg::MultiArrayDimension {
        label: crate::ros2_bridge::mappers::convert::to_ros_string(bus.label),
        size: bus.size,
        stride: bus.stride,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct StdMsgsMultiArrayDimensionMapper;

impl TypedTopicMapper for StdMsgsMultiArrayDimensionMapper {
    type Ros = ros_env::std_msgs::msg::MultiArrayDimension;
    type Bus = crate::std_msgs::msg::v1::MultiArrayDimension;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(multi_array_dimension_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(multi_array_dimension_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/std_msgs/multi_array_dimension.py
"""Generated mapper for `std_msgs/msg/MultiArrayDimension`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert


def multi_array_dimension_to_bus(msg):
    from robot_bus.std_msgs.msg.v1 import MultiArrayDimension as BusMsg

    bus = BusMsg()
    bus.label = str(msg.label)
    bus.size = msg.size
    bus.stride = msg.stride
    return bus


def multi_array_dimension_to_ros(bus):
    from std_msgs.msg import MultiArrayDimension as RosMsg

    out = RosMsg()
    out.label = str(bus.label)
    out.size = bus.size
    out.stride = bus.stride
    return out


class StdMsgsMultiArrayDimensionMapper:
    def type_name(self) -> str:
        return "std_msgs/msg/MultiArrayDimension"

    def ros_msg_type(self):
        from std_msgs.msg import MultiArrayDimension as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return multi_array_dimension_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.std_msgs.msg.v1 import MultiArrayDimension as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return multi_array_dimension_to_ros(bus)
